### PR TITLE
[5.6] [stdlib] Some fixes for scalar names

### PR DESCRIPTION
**Pull request by Azoy**
This cherry picks the changes made to fix getting some scalar names.
(Note this is dependent on https://github.com/apple/swift/pull/40758 )

(Cherry pick of https://github.com/apple/swift/pull/40644 for 5.6)

### CODE REVIEW

**Comment by Azoy** [comment]
@swift-ci please test

**Comment by Azoy** [comment]
@swift-ci please clean test macOS

**Review by stephentyrone — APPROVED** [review]

**Comment by Azoy** [comment]
@swift-ci please test

File: utils/gen-unicode-data/Sources/GenScalarProps/Names.swift [hunk 1]
```diff
@@ -159,6 +159,12 @@ func emitScalarNames(
     }
   }
   
+  result += """
+  #define NAMES_LAST_SCALAR_OFFSET \(nameBytes.count)
```
Comment (stephentyrone) [inline utils/gen-unicode-data/Sources/GenScalarProps/Names.swift:163]: Normally in C I would define these in terms of the collection:
```
static const size_t _swift_stdlib_names_last_offset = sizeof _swift_stdlib_names / sizeof _swift_stdlib_names[0];
```

Given that they're programmatically generated, I'm ok with doing it they way you have it, so no need to fix this, but I generally prefer this style.

**Comment by Azoy** [comment]
@swift-ci please clean test macOS

**Pull request merged**